### ISSUE TITLE
declutter rule WEIGHT line does not work for PyQt6

### ISSUE THREAD

**Issue by jknGH** [leo-editor/leo-editor#4141]
a declutter rule including a line like

`WEIGHT Bold`

has no effect in PyQt6. (a) the weight scaling seems to have changed (1--1000 instead of 1--99), and (b) the placement of Weight within QFont may have changed.

I will be submitting a Pull Request to fix this shortly

**Comment by jknGH** [comment]
Hi @edreamleo - I tried to push a branch to address this and create a Pull Request, but I am getting a 403 error. I recently created a GitHub personal access token. Do I need to be granted permission to push a branch, or similar? Thanks, Jon N

![image](https://github.com/user-attachments/assets/a4acfb99-980d-4432-ae6b-6ee5a486cce5)

**Comment by edreamleo** [comment]
@jknGH I've just sent an invitation for you to become a member.

**Comment by jknGH** [comment]
Thanks, accepted ;-)

**Comment by tbpassin** [comment]
The way I've been working is that I have my own GitHub account and it contains a clone/fork of the Leo repo.  I do all my work in a branch in my own repo.  When I push the branch from my computer to GitHub, GH offers to create a PR in the Leo repo.

**Comment by edreamleo** [comment]
@jknGH You're welcome and welcome aboard.

**Comment by edreamleo** [comment]
@tbpassin @jknGH Thomas's work flow has caused problems in the past and may do so in the future.

Instead, I strongly suggest you use the standard way of collaborating:
```console
git clone https://github.com/leo-editor/leo-editor.git
```

**Comment by edreamleo** [comment]
@jknGH @tbpassin After cloning Leo's repo, you **must** base all your work on Leo's "devel" branch:
```console
cd <your cloned repo>
git checkout devel
git checkout -b <your branch name>
(do your work)
git add .
git commit -m "Your commit message"
git push origin <your branch name>
```
A good format for branch name is: `jknGH-<issue number>-<short description of issue>`

**Update** The first time you push to origin you will see a link in the console. Click on that link to create a PR.

HTH.

**Comment by jknGH** [comment]
I'm not 100% sure I am seeing the difference between what @edreamleo and @tbpassin are saying here re. workflow.

I basically did this (like in my day job ;-)

`
# I already have devel locally
$ git switch -c jknGH-new-branch-name    # same effect as Edward's commands
# do my work
# ...
$ git push # git reminds me to use 'origin branch-name'
# follow link to create Pull Request (I use Gitlab more, where the terminology is 'merge request')
`

That seems to be the same as Edward ... but I am not clear how it differs from Thomas' description. He says 'clone/fork'; so he is trying to do PRs from a 'mostly-identical' repo held on his own account onto this one?

I would not do that ;-)

**Comment by edreamleo** [comment]
@jknGH Seems like you know what you're doing. Go for it :-)

**Comment by tbpassin** [comment]
> Instead, I strongly suggest you use the standard way of collaborating:
> 
> ```
> git clone https://github.com/leo-editor/leo-editor.git
> ```

Sure, that's how I got my clone.

**Comment by tbpassin** [comment]
I wrote "clone/fork" only because cloning is sometimes called a "forking".  Of course, I'm not forking Leo to take it some down other path. My clone repo only different in that I have my own branches for developing changes, whose names I prefix with "tbp-".  For me, "origin" is my clone on GitHub, "upstream" is Leo's own repo on GitHub. I push to origin, and GitHub notices and offers to create a PR in Leo's repo.  I don't know what is non-standard about any of this.

**Comment by jknGH** [comment]
> I wrote "clone/fork" only because cloning is sometimes called a "forking". Of course, I'm not forking Leo to take it some down other path. My clone repo only different in that I have my own branches for developing changes, whose names I prefix with "tbp-". For me, "origin" is my clone on GitHub, "upstream" is Leo's own repo on GitHub. I push to origin, and GitHub notices and offers to create a PR in Leo's repo. I don't know what is non-standard about any of this.

I have no experience of the last part (Gitlab offering to create a PR from your 'forked' repo, also hosted on GitHub). Sounds like that is the part giving Edward trouble?

**Comment by edreamleo** [comment]
@jknGH It is Thomas who has been having trouble with PRs. Let's move on.

**Comment by tbpassin** [comment]
> I have no experience of the last part (Gitlab offering to create a PR from your 'forked' repo, also hosted on GitHub). Sounds like that is the part giving Edward trouble?

Nothing is giving Edward trouble here, and I don't have trouble creating a PR.  Some time ago devel had been massively updated after I created the branch that eventually became became my PR.  I didn't know that I could pull from devel into my branch without messing up my new work, and the PR would have created a massive merge headache. Now I know.

There have been a few other times of lesser impact as I've learned how to work with the system.  It seems that I don't have good intuition about some operations and terminology  that seemed obvious to git's developers.
